bftengine/src/preprocessor/RequestProcessingState.cpp

```diff
@@ -159,10 +163,14 @@ PreProcessingResult RequestProcessingState::definePreProcessingConsensusResult()
     if (primaryPreProcessResultLen_ != 0 && !retrying_) {
       // Primary replica calculated hash is different from a hash that passed pre-execution consensus => we don't have
       // correct pre-processed results. Let's launch a pre-processing retry.
+      const auto &primaryHash =
```

Comment:
Taking a reference to a temporary is kinda weird. Maybe remove the `&` so the move constructor is used. I'm not sure this is actually safe as is.

Reply:
Why not safe? The variables are in use only inside the scope. But yes, the reference is not needed.

Reply:
Yeah, you are right. It's fine. Not sure what I was thinking.